### DISCUSSION TITLE
Is there a way to implement consume acknowledgement sync when using Classic Queue in federation mode

### DISCUSSION THREAD

**Discussion by mmpei** [rabbitmq/rabbitmq-server#6765] [Questions] [answered]
Now we are mainly using classic queue. and also we'd like to build a active-standby HA mode in two DCs.
we have read a lot, and know the federation mode by official and the another VMware designed but the second one not support classic queue.
and i also know the difficulty to sync consume acks between two clusters,
1. there is no api to get the one message and flag it.
2. there is no a file like wal to record the acks in classic queue, so it's difficult to replicate it.

so any solution?

**Comment by michaelklishin** [comment, marked as answer]
Federation does not propagate acknowledgement state because

 * Exchange federation *replicates* messages to external clusters
 * Queue federation moves them using a regular consumer (and client) with a low priority

Warm standby replication in VMware RabbitMQ replicates full quorum queue state. It does not support classic queues because classic queue replication **will be removed** in RabbitMQ 4.0 and it does not make sense to replicate a queue that's not replicated within a cluster to external clusters.

**Reply by mmpei** [reply]
Thank you michael.
Could you please give us some design suggestion or reference if we'd like to achieve active-standby mode in classic queues?
